### PR TITLE
XW-1124 | Bump eslint version and fix all errors

### PR DESCRIPTION
**Pull request by vforge**
/cc @Wikia/x-wing

### CODE REVIEW

File: front/common/baseline/tracker-universal-analytics.js [hunk 1]
```diff
@@ -433,7 +433,7 @@ if (typeof window.M.tracker === 'undefined') {
 		initAccount(accountPrimary, domain);
 		initAccount(accountAds, domain);
 
-		if (Boolean(M.prop('isGASpecialWiki') || Mercury.wiki.isGASpecialWiki)) {
+		if (M.prop('isGASpecialWiki') || Mercury.wiki.isGASpecialWiki) {
```
Comment (hakubo) [inline front/common/baseline/tracker-universal-analytics.js]: why do we need to change that?

File: server/app/facets/auth/birthdate-input.js [hunk 2]
```diff
@@ -101,6 +101,7 @@ export default class BirthdateInput {
 				return [data.month, data.day, data.year];
 			case 'little':
 				// covered by default block
+				// eslint-disable no-fallthrough
```
Comment (hakubo) [inline server/app/facets/auth/birthdate-input.js]: instead of working around 
do we have to keep case 'little' - is it more readable?

Comment (bognix) [inline server/app/facets/auth/birthdate-input.js]: agree with @hakubo. I think we can achieve the same level of readability by adding comment to default

File: .eslintrc [hunk 3]
```diff
@@ -194,7 +194,7 @@
     "no-var": 2,
     "object-shorthand": 2,
     "prefer-arrow-callback": 2,
-    "prefer-const": 2,
+    "prefer-const": 0,
```
Comment (bognix) [inline .eslintrc]: why we are disabling this rule?

File: front/common/baseline/tracker-universal-analytics.js (hunk 1 above)
Comment (vforge) [inline front/common/baseline/tracker-universal-analytics.js]: 1) Conditional casts everything to `boolean` anyway.
2) "no-extra-boolean-cast" - this rule told us this.

File: .eslintrc (hunk 3 above)
Comment (vforge) [inline .eslintrc]: Talked offline

**Comment by bognix** [comment]
LGTM

**Comment by rogatty** [comment]
LGTM

**Pull request merged**